### PR TITLE
Add signed in state to evergreen WNP [fix #9245]

### LINKED ISSUE #9245: Promote Firefox mobile to signed-in users on the fallback WNP
## Description

The new fallback whatsnew page created in https://github.com/mozilla/bedrock/issues/8126 promotes a Firefox Account, but shows the same CTA even to users who are signed in. We'll add a second conditional state showing the send-to-device widget to users who are already signed into Firefox.

---

#### Success Criteria
- [ ] The page is updated
- [ ] Signed out users get the account CTA
- [ ] Signed in users get the send-to-device widget

#### Approvals
- [ ] The Accountable
- [ ] Brand - design?
- [ ] Brand - copy?
- [ ] Team Lead
- [ ] Others?

### PR DESCRIPTION
**Pull request by craigcook**
## Description
Adds a signed-in state to the evergreen fallback WNP, shown to locales that haven't translated the current version's custom page or to any users updating to an older version that doesn't have an active custom page.

If the user is not signed into sync, we display a pitch to create an account (or sign in).
If the user is signed into sync, we display the send-to-device widget to promote Firefox mobile.

This adds a few new strings to `/firefox/whatsnew/whatsnew-account.ftl`

## Issue / Bugzilla link
#9245 

## Testing
http://localhost:8000/firefox/60.0/whatsnew/all/
http://localhost:8000/firefox/60.0/whatsnew/all/?signed-in=true

https://www-demo-pr-9276.herokuapp.com/firefox/60.0/whatsnew/all/
https://www-demo-pr-9276.herokuapp.com/firefox/60.0/whatsnew/all/?signed-in=true

- [x] Account promo is shown when signed out
- [x] Mobile widget is shown when signed in
- [x] Tests pass

### CODE REVIEW

**Comment by craigcook** [comment]
Marking Do Not Merge for the moment until I can get stakeholder approval.

**Comment by craigcook** [comment]
@peiying2 Could you review the string additions here? Thanks!

File: bedrock/firefox/templates/firefox/whatsnew/index-account.html [hunk 1]
```diff
@@ -36,22 +41,52 @@
...
+          </p>
+        </div>
+
+      {# If the user is signed in, promote mobile #}
+        <div class="show-fxa-supported-signed-in">
+          <img class="c-emphasis-box-logo" src="{{ static('protocol/img/logos/firefox/browser/logo-md.png') }}" alt="">
+        {# If user is in a locale with translated basket messages... #}
+        {% if show_send_to_device %}
+          <h2 class="c-emphasis-box-title">{{ ftl('whatsnew-account-signed-in-headline') }}</h2>
+          {# L10n: "You got it" here is a casual answer to the previous question, "Want privacy on every device?" #}
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/index-account.html:71]: L10n comments should live in .ftl files and not in .html. Is this relevant to the string in use, or copy pasta maybe?

File: bedrock/firefox/templates/firefox/whatsnew/index-account.html [hunk 2]
```diff
@@ -36,22 +41,52 @@
...
+            ) }}
+          </p>
+        </div>
+
+      {# If the user is signed in, promote mobile #}
+        <div class="show-fxa-supported-signed-in">
+          <img class="c-emphasis-box-logo" src="{{ static('protocol/img/logos/firefox/browser/logo-md.png') }}" alt="">
+        {# If user is in a locale with translated basket messages... #}
+        {% if show_send_to_device %}
+          <h2 class="c-emphasis-box-title">{{ ftl('whatsnew-account-signed-in-headline') }}</h2>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/index-account.html:70]: Could we fall back to the default S2D title here, if a locale doesn't have it translated?

File: media/js/firefox/whatsnew/whatsnew-account.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    var client = Mozilla.Client;
+    var sendTo = document.getElementById('send-to-device');
+
+    if (sendTo) {
+        var form = new Mozilla.SendToDevice();
+        form.init();
+    }
+
+    // bug 1419573 - only show "Your Firefox is up to date" if it's the latest version.
```
Comment (alexgibson) [inline media/js/firefox/whatsnew/whatsnew-account.js:21]: Is this needed? The bundle already seems to use `whatsnew/up-to-date.js`.

File: media/js/firefox/whatsnew/whatsnew-account.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// create namespace
+if (typeof window.Mozilla === 'undefined') {
+    window.Mozilla = {};
+}
+
+(function($, Mozilla) {
```
Comment (alexgibson) [inline media/js/firefox/whatsnew/whatsnew-account.js:10]: Nit: remove `$`

File: media/js/firefox/whatsnew/whatsnew-account.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    // bug 1419573 - only show "Your Firefox is up to date" if it's the latest version.
+    if (client.isFirefoxDesktop) {
+        client.getFirefoxDetails(function(data) {
+            if (data.isUpToDate) {
+                document.querySelector('.c-page-header').classList.add('show-up-to-date-message');
+            }
+        });
+    }
+
+})(window.jQuery, window.Mozilla);
```
Comment (alexgibson) [inline media/js/firefox/whatsnew/whatsnew-account.js:30]: Nit: remove `window.jQuery`

File: tests/functional/firefox/whatsnew/test_whatsnew_60.py [hunk 6]
```diff
@@ -12,3 +13,22 @@
 def test_account_buttons_displayed(base_url, selenium):
     page = FirefoxWhatsNew60Page(selenium, base_url, params='').open()
     assert page.is_account_button_displayed
+
+
+@pytest.mark.skip_if_not_firefox(reason='Whatsnew pages are shown to Firefox only.')
+@pytest.mark.nondestructive
+def test_send_to_device_success(base_url, selenium):
```
Comment (alexgibson) [inline tests/functional/firefox/whatsnew/test_whatsnew_60.py:20]: Worth adding another test for the QR code that some locales see?

File: bedrock/firefox/templates/firefox/whatsnew/index-account.html (hunk 1 above)
Comment (craigcook) [reply bedrock/firefox/templates/firefox/whatsnew/index-account.html:71]: T'was pasta, will remove.

**Comment by craigcook** [comment]
Successful test run: https://gitlab.com/mozmeao/www-config/-/pipelines/176327593
